tests/test_helpers.py

```diff
@@ -515,6 +515,71 @@ def test_proxies_from_env_http_with_auth(mocker) -> None:
     assert proxy_auth.encoding == "latin1"
 
 
+# --------------------- get_env_proxy_for_url ------------------------------
+
+
+def test_get_env_proxy_for_url_when_url_matches_the_no_proxy_list(mocker) -> None:
+    mocker.patch("aiohttp.helpers.proxy_bypass", return_value=True)
+    mocker.patch(
+        "aiohttp.helpers.proxies_from_env",
+        return_value={"http": helpers.ProxyInfo("http://example.com", "")},
+    )
+    url = URL("http://aiohttp.io/path")
+    with pytest.raises(LookupError, match=r"Proxying is disallowed for `'aiohttp.io'`"):
+        helpers.get_env_proxy_for_url(url)
+
+
+def test_get_env_proxy_for_url_when_url_scheme_does_not_match_proxy_list(
+    mocker,
+) -> None:
+    mocker.patch("aiohttp.helpers.proxy_bypass", return_value=False)
+    proxies = {"https": helpers.ProxyInfo("https://example.com", "")}
+    mocker.patch("aiohttp.helpers.proxies_from_env", return_value=proxies)
+    url = URL("http://aiohttp.io/path")
+
+    with pytest.raises(
+        LookupError, match=r"No proxies found for `http://aiohttp.io/path` in the env"
+    ):
+        helpers.get_env_proxy_for_url(url)
+
+
+def test_get_env_proxy_for_url_when_url_is_empty(
+    mocker,
+) -> None:
+    mocker.patch("aiohttp.helpers.proxy_bypass", return_value=False)
+    proxies = {"https": helpers.ProxyInfo("https://example.com", "")}
+    mocker.patch("aiohttp.helpers.proxies_from_env", return_value=proxies)
+    url = URL("")
+
+    with pytest.raises(LookupError, match=r"No proxies found for `` in the env"):
+        helpers.get_env_proxy_for_url(url)
+
+
+@pytest.mark.parametrize(
+    ("schema", "proxyInfo", "url"),
+    [
+        (
+            "https",
+            helpers.ProxyInfo("https://other.com", ""),
+            URL("https://aiohttp.com"),
+        ),
+        (
+            "http",
+            helpers.ProxyInfo("http://example.com", ""),
+            URL("http://aiohttp.com"),
+        ),
+    ],
```

Comment:
When the params are complex or even just non-scalar, pytest tries to stringify them for the report and they don't look helpful. So it's always best to provide explicit IDs in such cases:
```suggestion
    ],
    ids=("https", "http"),
```

Besides, one can use those params to invoke a specific test as follows:
```console
$ pytest 'tests/test_helpers.py::test_get_env_proxy_for_url_when_url_scheme_matches_proxy_list[https]'
```
(it'd be less trivial with a complex structure which would need the right amount of escaping things)